File: Javascript/quicksort.js
// Input array that has to be sorted
var n = [5,3,7,6,2,9];

// Swap Function
function swap(n, l, r){
    var temp = n[l];
    n[l] = n[r];
    n[r] = temp;
}

// Paritioning of array ,so that the elements can be sorted
function part(n, l, r) {
    var pivot   = n[Math.floor((r + l) / 2)], 
        i       = l,  j       = r; 
    while (i <= j) {
        while (n[i] < pivot) {
            i++;
        }
        while (n[j] > pivot) {
            j--;
        }
        if (i <= j) {
            swap(n, i, j); 
            i++;
            j--;
        }
    }
    return i;
}
 // Sorting function
function Sort(n, l, r) {
    var x;
    if (n.length > 1) {
        x = part(n, l, r); 
        if (l < x - 1) { 
            Sort(n, l, x - 1);
        }
        if (x < r) { 
            Sort(n, x, r);
        }
    }
    return n;
}
// 
var sortedArray = Sort(n, 0, n.length - 1);
console.log(sortedArray); 

File: C++/quicksort.cpp
#include<iostream>

using namespace std;

int partition(int A[], int l, int h)
{
    int temp;
    int pivot = A[l];
    int i = l, j = h;
    do
    {
        do
        {
            i++;
        } while (A[i] >= pivot);
        do
        {
            j--;
        } while (A[j] < pivot);
        if (i < j){
            temp = A[i];
            A[i] = A[j];
            A[j] = temp;
        }
    } while (i < j);
    temp = A[l];
    A[l] = A[j];
    A[j] = temp;
    return j;
}
void QuickSort(int A[], int l, int h)
{
    int j;
    if (l < h)
    {
        j = partition(A, l, h);
        QuickSort(A, l, j);
        QuickSort(A, j + 1, h);
    }
}
int main()
{
    int size;
    cin >> size;
    int A[size];
    for(int i=0;i<size;i++){
        cin >> A[i];
    }
    QuickSort(A,0, size-1);
    for (int i = size-1; i >=0 ; i--)
        cout << A[i] << " ";
}